rts/System/Threading/ThreadPool.cpp

```diff
@@ -586,7 +586,9 @@ void SetMaximumThreadCount()
 
 void SetDefaultThreadCount()
 {
-	std::uint32_t systemCores  = springproc::CPUID::GetInstance().GetAvailableProceesorAffinityMask();
+	// The Hyper Threading pattern is to select only one hardware thread per core.
+	std::uint32_t systemCores  = springproc::CPUID::GetInstance().GetAvailableProceesorAffinityMask()
+								& ( Threading::HasHyperThreading() ? 0x55555555 : 0xffffffff );
```

Comment:
The illustration from the PR description sounds like an excellent piece of documentation to put here:
```suggestion
	/* The logical to physical processor map usually looks like this
	 * when hyperthreaded, with adjacent logicals on the same physical:
	 * | **----------  Physical Processor 0 (Hyperthreaded)
	 * | --**--------  Physical Processor 1 (Hyperthreaded)
	 * | ----**------  Physical Processor 2 (Hyperthreaded)
	 * | ------**----  Physical Processor 3 (Hyperthreaded)
	 * | --------**--  Physical Processor 4 (Hyperthreaded)
	 * | ----------**  Physical Processor 5 (Hyperthreaded)
	 * This mask ensures one hardware thread per physical core. */
	static constexpr auto HYPERMASK = 0b01010101010101010101010101010101;

	std::uint32_t systemCores  = springproc::CPUID::GetInstance().GetAvailableProceesorAffinityMask()
								& ( Threading::HasHyperThreading() ? HYPERMASK : 0xffffffff );
```

Reply:
That's a good call.